app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -611,8 +612,24 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 refreshCurrentView(false);
                 break;
         }
+        mFormController.setPendingCalloutFormIndex(null);
     }
-    
+
+    private IntentWidget getPendingIntentWidget() {
+        IntentWidget bestMatch = null;
+
+        //Ugh, copied from the odkview mostly, that's stupid
+        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
+            //Figure out if we have a pending intent widget
+            if (q instanceof IntentWidget) {
+                if(((IntentWidget) q).getFormId().equals(mFormController.getPendingCalloutFormIndex())) {
```

Comment:
It would be good to generalize this process into the intent callout setup above and make it usable by Image Capture, etc. 

Don't have to do in this PR, but maybe a TODO: or a comment would be a helpful hint to someone in the future that this is mostly done and just needs to get wired up